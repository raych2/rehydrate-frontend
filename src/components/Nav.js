import { Link } from 'react-router-dom';
import logo from '../assets/Rehydrate_Final_white_small2.png';

const Nav = () => {
  return (
    <nav>
      <Link to="/" className="logoContainer navlink">
        <img className="navLogo" alt="rehydrate logo" src={logo} />
        <div>Rehydrate</div>
      </Link>
      <ul className="navlinkContainer">
        <Link to="about" className="navlink">
          About
        </Link>
        <Link to="tips" className="navlink">
          Tips
        </Link>
        <Link to="stats" className="navlink">
          Stats
        </Link>
      </ul>
    </nav>
  );
};

export default Nav;
