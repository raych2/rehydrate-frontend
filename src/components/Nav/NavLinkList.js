import { Link } from 'react-router-dom';

const NavLinkList = ({ open, setOpen }) => {
  return (
    <ul
      className={open ? 'navlinkContainer' : 'navlinkContainer active'}
      onClick={() => setOpen(!open)}
    >
      <Link to="about" className="navlink">
        About
      </Link>
      <Link to="tips" className="navlink">
        Tips
      </Link>
      <Link to="stats" className="navlink">
        Stats
      </Link>
    </ul>
  );
};
export default NavLinkList;
