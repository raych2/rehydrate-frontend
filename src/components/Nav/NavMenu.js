import { useState } from 'react';
import NavLinkList from './NavLinkList';

const NavMenu = () => {
  const [open, setOpen] = useState(false);

  return (
    <>
      <div
        className={open ? 'menu' : 'menu close'}
        onClick={() => setOpen(!open)}
      >
        <div />
        <div />
        <div />
      </div>
      <NavLinkList open={open} setOpen={setOpen} />
    </>
  );
};

export default NavMenu;
