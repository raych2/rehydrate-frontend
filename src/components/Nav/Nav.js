import { Link } from 'react-router-dom';
import logo from '../../assets/Rehydrate_Final_white_small2.png';
import NavMenu from './NavMenu';

const Nav = () => {
  return (
    <nav>
      <Link to="/" className="logoContainer navlink">
        <img className="navLogo" alt="rehydrate logo" src={logo} />
        <div>Rehydrate</div>
      </Link>
      <NavMenu />
    </nav>
  );
};

export default Nav;
