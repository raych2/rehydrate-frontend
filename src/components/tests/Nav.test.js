import { render, screen } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import Nav from '../Nav/Nav';

describe('Nav component', () => {
  test('renders correctly', () => {
    const { container } = render(<Nav />, { wrapper: MemoryRouter });
    expect(container).toMatchSnapshot();
  });
  test('should have correct href for each page', () => {
    render(<Nav />, { wrapper: MemoryRouter });
    const homeLink = screen.getByRole('link', {
      name: 'rehydrate logo Rehydrate',
    });
    expect(homeLink.getAttribute('href')).toBe('/');
    const aboutLink = screen.getByRole('link', { name: 'About' });
    expect(aboutLink.getAttribute('href')).toBe('/about');
    const tipsLink = screen.getByRole('link', { name: 'Tips' });
    expect(tipsLink.getAttribute('href')).toBe('/tips');
    const statsLink = screen.getByRole('link', { name: 'Stats' });
    expect(statsLink.getAttribute('href')).toBe('/stats');
    const quizLink = screen.getByRole('link', { name: 'Quiz' });
    expect(quizLink.getAttribute('href')).toBe('/quiz');
  });
});
