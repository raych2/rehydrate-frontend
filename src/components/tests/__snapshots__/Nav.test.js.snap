// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav component renders correctly 1`] = `
<div>
  <nav>
    <a
      class="logoContainer navlink"
      href="/"
    >
      <img
        alt="rehydrate logo"
        class="navLogo"
        src="Rehydrate_Final_white_small2.png"
      />
      <div>
        Rehydrate
      </div>
    </a>
    <div
      class="menu close"
    >
      <div />
      <div />
      <div />
    </div>
    <ul
      class="navlinkContainer active"
    >
      <li>
        <a
          class="navlink"
          href="/about"
        >
          About
        </a>
      </li>
      <li>
        <a
          class="navlink"
          href="/tips"
        >
          Tips
        </a>
      </li>
      <li>
        <a
          class="navlink"
          href="/stats"
        >
          Stats
        </a>
      </li>
      <li>
        <a
          class="navlink"
          href="/quiz"
        >
          Quiz
        </a>
      </li>
    </ul>
  </nav>
</div>
`;
